DashKit/DashPumpManager.swift

```diff
@@ -0,0 +1,911 @@
+            }
+        }
+    }
+
+    public init(state: DashPumpManagerState, podCommManager: PodCommManagerProtocol = PodCommManager.shared) {
+        self.lockedState = Locked(state)
+        self.podCommManager = podCommManager
+        self.podCommManager.delegate = self
+
+        podCommManager.setLogger(logger: self)
+
+        podCommManager.enableAutoConnection(launchOptions: [:])
+    }
+
+    public convenience required init?(rawState: PumpManager.RawStateValue) {
+        guard let state = DashPumpManagerState(rawValue: rawState) else
+        {
+            return nil
+        }
+
+        self.init(state: state)
+    }
+
+    public var rawState: PumpManager.RawStateValue {
+        return state.rawValue
+    }
+
+    public var debugDescription: String {
+        var lines = [
+            "## DashPumpManager",
+            "* podCommState: \(podCommManager.podCommState)",
+            "* podId: \(String(describing: podCommManager.getPodId()))",
+        ]
+        
+        let semaphore = DispatchSemaphore(value: 0)
+        podCommManager.getPodStatus(userInitiated: false) { (result) in
+            switch result {
+            case .failure(let error):
+                lines.append("* podStatus: Error: \(error)")
+            case .success(let status):
+                lines.append(contentsOf: [
+                    "* activeAlerts: \(status.activeAlerts)",
+                    "* bolusRemaining: \(status.bolusRemaining)",
+                    "* bolusUnitsRemaining: \(status.bolusUnitsRemaining)",
+                    "* delivered: \(status.delivered)",
+                    "* expirationDate: \(status.expirationDate)",
+                    "* isOcclusionAlertActive: \(status.isOcclusionAlertActive)",
+                    "* podState: \(status.podState)",
+                    "* programStatus: \(status.programStatus)",
+                    "* reservoir: \(status.reservoir)",
+                    "* reservoirUnitsRemaining: \(status.reservoirUnitsRemaining)",
+                    "* totalUnitsDelivered: \(status.totalUnitsDelivered)",
+                ])
+            }
+            semaphore.signal()
+        }
+        let result = semaphore.wait(timeout: .now() + .seconds(8))
+        if case .timedOut = result {
+            lines.append("* podStatus: timed out")
+        }
+        lines.append(contentsOf: [
+            state.debugDescription,
+            "",
+        ])
+        return lines.joined(separator: "\n")
+    }
+}
+
+// Capture dash logs
+extension DashPumpManager: LoggingProtocol {
+    public func info(_ message: String) {
+        log.default("PodSDK Info: %{public}@", message)
+    }
+
+    public func debug(_ message: String) {
+        log.default("PodSDK Debug: %{public}@", message)
+    }
+
+    public func error(_ message: String) {
+        log.default("PodSDK Error: %{public}@", message)
+    }
+}
+
+extension DashPumpManager: PodCommManagerDelegate {
+    public func onAlert(alerts: PodAlerts) {
```

Comment:
Do we need to do anything more with these alerts, alarms, etc.?

Reply:
Yes, that's a future ticket.